test/testdata/core/types/subtyping_shapes.rb

```diff
@@ -0,0 +1,13 @@
+# typed: true
+extend T::Sig
+
+sig { params(x: T.any({a: Integer}, [String])).void }
+def bar(x)
+  T.reveal_type(x) # error: Revealed type: `T.any({a: Integer}, [String])`
+end
```

Comment:
Could you add another example where the arguments to `T.any` are flipped? I think that will exercise the missing symmetric case.

Reply:
Even though it shouldn't be different, would you mind adding some additional tests that exercise the case where the args to the `T.any` are flipped?